### PR TITLE
Convert HandlerStage to GraphStage, #19361.

### PR DESCRIPTION
**Pull request by danxmoran**
Refs #19361.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Thanks! Will review soon :)

File: akka-http-core/src/main/scala/akka/http/impl/engine/ws/FrameHandler.scala [hunk 1]
```diff
@@ -21,158 +23,163 @@ import scala.util.control.NonFatal
...
-      def handleRegularFrameStart(start: FrameStart)(implicit ctx: Ctx): SyncDirective =
-        (start.header.opcode, start.isFullMessage) match {
-          case (Opcode.Binary, true)  ⇒ publishMessagePart(BinaryMessagePart(start.data, last = true))
-          case (Opcode.Binary, false) ⇒ becomeAndHandleWith(new CollectingBinaryMessage, start)
-          case (Opcode.Text, _)       ⇒ becomeAndHandleWith(new CollectingTextMessage, start)
-          case x                      ⇒ protocolError()
+    override def createLogic(attributes: Attributes): GraphStageLogic =
+      new GraphStageLogic(shape) with OutHandler {
+        setHandler(out, this)
+        setHandler(in, IdleHandler)
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/ws/FrameHandler.scala]: 👍

**Comment by danxmoran** [comment]
Hey @ktoso, just a heads-up that I'm going abroad and won't have access to my dev environment for around a month starting this Sunday.  Do you have time to get this PR merged before then?  If not, should I push my branch to the central repo so somebody else can pick it up?  Don't want to leave another dead PR lying around.

**Comment by ktoso** [comment]
Thanks for the info, just about to sit down and spend the evening on merging things :)
Hope this one's good, will let you know or push it over the finish line 👍

**Comment by ktoso** [comment]
Reviewed and I think it's good. All states seem to be translated well, I'll have another look in the morning but this should definitely make it for the 2.4.8 tomorrow.

Thanks a lot, this was a large chunk of work.
And sorry reviewing was a bit slow here :)

**Comment by patriknw** [comment]
LGTM

**Comment by ktoso** [comment]
Thanks a lot @danxmoran !

**Pull request merged**